syntax = "proto3";

package transport_catalogue_pb;

import "map_renderer.proto";
import "transport_router.proto";

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    string          name           = 1;
    Coordinates     coordinates    = 2;
    repeated string routes_through = 3;
}

message Route {
    string          name     = 1;
    repeated string stops    = 2;
    bool            circular = 3;
}

message RouteStats {
    int32  stops_count        = 1;
    int32  unique_stops_count = 2;
    double actual_length      = 3;
    double curvature          = 4;
}

message Distanse {
    string from     = 1;
    string to       = 2;
    int32  distance = 3;
}

message TransportCatalogue {
    repeated Route routes                                = 1;
    repeated Stop stops                                  = 2;
    repeated Distanse stop_distances                     = 3;
    transport_router_pb.TransportRouter transport_router = 4;
    map_pb.OutputSettings output_settings                = 5;
}